docs/delete-product-or-product-variant-metafields-in-bulk/script.liquid

```diff
@@ -0,0 +1,221 @@
+{% if options.delete_product_metafields__boolean and options.delete_product_variant_metafields__boolean %}
+  {% error "You can only choose one option to target, either product metafields or product variant metafields, not both at the same time." %}
+{% endif %}
+
+{% if options.delete_product_metafields__boolean == false and options.delete_product_variant_metafields__boolean == false %}
+  {% error "You must choose one option to target, either product metafields or product variant metafields." %}
+{% endif %}
+
+{% if options.metafield_namespace == blank and options.metafield_key != blank %}
+  {% error "If you provide a metafield key, you must also provide a metafield namespace." %}
+{% endif %}
+
+{% if event.topic == "mechanic/user/trigger" %}
+{% comment %} Delete product metafields query {% endcomment %}
+  {% if options.delete_product_metafields__boolean %}
+    {% capture bulk_operation_query %}
+        query {
+          products {
```

Comment:
It looks like the following section here is identical to the `productVariants` section below. That's ~40 lines of duplicated code - can you de-dup this, replacing the `products` connection here with something like...

```liquid
{% if options.delete_product_metafields__boolean %}
  products
{% elsif options.delete_product_variant_metafields__boolean %}
  productVariants
{% endif %}
```

?

Reply:
Thanks for the suggestion! I changed add my commits.